### ISSUE TITLE
Changing local state refetches a query using it as @client @export only once on v2.6.4

### ISSUE THREAD

**Issue by NinjaOnRails** [apollographql/apollo-client#5353]
#4530 Is closed so I reopened a similar issue.
Following Apollo Client's official docs [here](https://www.apollographql.com/docs/react/essentials/local-state/#using-client-fields-as-variables), using local state field as variable of a remote query with `@client @export` should refetch it every time the local state state field changes. This finally happens in v2.6.4, but only once. It won't do it the second time local state field changes. I deployed a live demo [here](https://apollo-localstate-export.danni.now.sh/) and code is on github [here](https://github.com/NinjaOnRails/apollo-localstate-bug/blob/master/pages/index.js). After the third or fourth time local state field is changed, it cannot be changed anymore, which is even stranger.

I've tried methods of refetching other than `@client @export` such as refetching the query the usual way by adding `refetchQueries` to the local mutation HOC or hook with variables with values of the latest corresponding local state field. Passing local state field as variable to separate query calls through `client.query()`, HOC and hooks has the same result as this. It kind of works, but it has to be called twice after the first time. The first time it works, the second time it has to be called twice to work, then works again, then has to be called twice again, and then it just keeps repeating like that. You can see the behaviour on my app in production [here](https://www.danni.tv/watch?id=cjzg1ocza01f607444lwr9sui&audioId=cjzg1od9s01g407440f3o9dw3). Try toggling the languages(country flags on any of the video watch pages). I had to add a page reload on the home page after each local state change, which is bad user experience for now

**Comment by NinjaOnRails** [comment]
[Here's the code](https://github.com/NinjaOnRails/apollo-localstate-not-updating/blob/master/pages/index.js) for the problem described in my second paragraph. It looks like by re-running the remote query with variables manually after the Local State change, it stops the Local State from updating itself altogether, which results in the behaviour described. It doesn't matter whether I change the Local State with HOC mutation and resolver and then refetch the query by passing in `refetchQueries` with variables or with `client.writeData()` and then refetch manually with HOC, `client.query` or `refetchQueries`, the result is the same. I guess I'll try to use the component state as a variable as a temporary fix for now.

```
const onClick = async ({ target: { id } }, client, contentLanguage) => {
  contentLanguage.push(id);
  await client.writeData({
    data: { contentLanguage },
  });
  await client.query({ query: LOCAL_STATE });
  client.query({ query: AUDIOS_QUERY, variables: { contentLanguage } });
};
```

**Comment by hwillson** [comment]
A lot of the Apollo Client internals have changed since v3 was launched. We recommend trying a more modern version of `@apollo/client`. Let us know if you're still encountering this issue. Thanks!
